lib/fail.js

```diff
@@ -3,6 +3,7 @@ var match = require('pattern-match');
 function ValidationError(message, stack) {
     Error.call(this, message);
     this.stack = stack;
+    this.message = message;
```

Comment:
Ah. I was previously unaware that creating custom Error subclasses in Javascript is complicated, but having now read some documentation about it, it seems so.

The code in the tree right now doesn't exactly follow the pattern recommended (on this page)[https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Error], but your patch gets us closer, so it looks good to me, for now.